# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

I choose Lane 2: Refresh / Content Opportunity Scoring.

I chose this lane because content teams usually have limited time and need to decide which pages should be reviewed first. Instead of manually checking thousands of pages, this project aims to rank content items using observable search and engagement signals.

The goal is not to automate content decisions completely, but to build a decision-support system that helps humans prioritize pages that may deserve attention.

## 2. The question: decision, action, cost of a wrong call

My research question is:

Which content pages should be reviewed first for refresh, improvement, monitoring, or other actions based on available search and engagement signals?

Decision:
The decision is which pages should receive attention from a content team first.

Action:
A content reviewer could inspect the ranked pages and decide whether to update content, improve metadata, monitor performance, or leave the page unchanged.

Cost of a wrong recommendation:
A false positive could waste time by sending reviewers to pages that do not need changes. A false negative could cause an important opportunity or declining page to be missed.

The system will provide recommendations for prioritization, not guaranteed predictions of which pages will improve.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
# ==========================================
# 3. Quick look at the data
# ==========================================

import pandas as pd
import os

# Clone repository if it is not already available
if not os.path.exists("ML1"):
    !git clone https://github.com/albijanashala/ML1.git

# Path to starter dataset
file_path = "ML1/data/raw/content_refresh_anonymized.csv"

# Load dataset
df = pd.read_csv(file_path)

# Show dataset size
print("Dataset shape:")
print(df.shape)

# Show columns
print("\nColumns:")
print(df.columns.tolist())

# Number of content items
print("\nNumber of content items:")
print(len(df))

# Average impressions
if "impressions_90d" in df.columns:
    print("\nAverage impressions (90 days):")
    print(round(df["impressions_90d"].mean(), 2))

# Pages with declining trend
if "trend_direction" in df.columns:
    declining_pages = (df["trend_direction"] == "down").sum()
    print("\nPages with declining trend:")
    print(declining_pages)

# Average sessions
if "sessions_90d" in df.columns:
    print("\nAverage sessions (90 days):")
    print(round(df["sessions_90d"].mean(), 2))

# Display first rows
print("\nFirst 5 rows:")
df.head()

Dataset shape:
(30000, 44)

Columns:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']

Number of content items:
30000

Average impressions (90 days):
5200.37

Pages with declining trend:
16262

Average sessions (90 days):
37.07

First 5 rows:


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


Based on the initial execution of the starter dataset, this lane presents a significant opportunity for the next 7 weeks due to these real-world baseline metrics:

Dataset Scale: We are working with a clean slice of 30,000 unique content items to build and test our initial models.

Active Performance Decay: Out of these rows, exactly 16,262 pages exhibit a trend_direction == "down" (not 13,327). This substantial volume confirms that manual review is impossible at scale and a data-backed ranking queue is critically needed.

High Exposure Risk: The average page captures 5,200.37 impressions (not 3,501) and 37.07 sessions over a 90-day window. Missing a decaying page or wasting time on an asset with no demand carries an expensive traffic and revenue penalty.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I CAN claim:

"This system serves strictly as an observational decision-support tool designed to optimize the allocation of limited human operational focus."

"The model successfully learns to rank historic performance degradation patterns based on real search signals like impressions, clicks, position drift, and user engagement."

What I CANNOT claim:

I cannot claim that an automated refresh recommendation guarantees traffic recovery or revenue increases (the dataset is purely observational, not a controlled causal experiment).

I cannot claim to have reverse-engineered, mapped, or predicted the underlying Google or search engine ranking factors.

I cannot claim any semantic or textual understanding of the assets, as all text, domains, raw URLs, and private queries are completely scrambled into pseudonymized hashes.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.